Core/AppURLs+StatisticsDependent.swift

```diff
@@ -0,0 +1,148 @@
+//
+//  AppURLs+StatisticsDependent.swift
+//  DuckDuckGo
+//
+//  Copyright © 2023 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import BrowserServicesKit
+
+public extension URL {
+
+    private static let defaultStatisticsDependentURL = StatisticsDependentURL()
+
+    static func makeSearchURL(text: String) -> URL? { defaultStatisticsDependentURL.makeSearchURL(text: text) }
+    static func makeSearchURL(query: String, queryContext: URL? = nil) -> URL? {
+        defaultStatisticsDependentURL.makeSearchURL(query: query, queryContext: queryContext)
+    }
+    static func applyingStatsParams(for url: URL) -> URL { defaultStatisticsDependentURL.applyingStatsParams(for: url) }
```

Comment:
This is slightly confusing, it looks like a factory method here, but I know how it's used.  Maybe this should be an instance func instead?

Reply:
makes sense! I changed hasCorrectMobileStatsParams too